R/rtf_strwidth.R

```diff
@@ -131,3 +132,37 @@ rtf_strwidth <- function(tbl) {
 
   width
 }
+
+#' Safe version of `graphics::par()` with side effects isolated
+#'
+#' @noRd
+safe_par <- function(...) with_bmp(graphics::par(...))
+
+#' Safe version of `graphics::strwidth()` with side effects isolated
+#'
+#' @noRd
+safe_strwidth <- function(text, ...) with_bmp(graphics::strwidth(text, ...))
+
+#' Evaluate expression within a BMP device and close it on exit
+#'
+#' See why: <https://github.com/Merck/r2rtf/issues/227>
+#'
+#' @noRd
+with_bmp <- function(expr) {
+  expr <- substitute(expr)
+  grDevices::bmp(file_null())
+  on.exit(
+    {
+      current <- grDevices::dev.cur()
+      if (current > 1) {
+        prev <- grDevices::dev.prev(current)
+        grDevices::dev.off(current)
+        if (prev != current) grDevices::dev.set(prev)
+      }
+    },
+    add = TRUE
+  )
+  eval(expr, envir = parent.frame())
+}
+
+file_null <- function() if (.Platform$OS.type == "windows") "nul:" else "/dev/null"
```

Comment:
Could we use "tempfile" to avoid a OS dependent logic?

Reply:
I guess writing to the null device is often faster than a real file.